File: SampleApp-Java/app/src/main/java/rishi/alphae/sampleapp/UserAdapter.java
package rishi.alphae.sampleapp;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import java.util.ArrayList;

/**
 * Created by rishi on 4/2/19.
 */

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.ViewHolder> {

    private ArrayList<User> mUsers;

    public UserAdapter(ArrayList<User> users) {
        super();
        mUsers = users;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.user_card, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(UserAdapter.ViewHolder holder, int position) {
        User user = mUsers.get(position);
        holder.userName.setText(user.getName());
        holder.userAge.setText(user.getAge().toString());
    }

    @Override
    public int getItemCount() {
        return mUsers.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        private TextView userName;
        private TextView userAge;

        public ViewHolder(View itemView) {
            super(itemView);
            userName = itemView.findViewById(R.id.user_name);
            userAge = itemView.findViewById(R.id.user_age);
        }
    }


}

File: SampleApp-Kotlin/app/src/main/java/rishi/alphae/sampleapp/UserAdapter.kt
package rishi.alphae.sampleapp

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

/**
 * Created by rishi on 3/2/19.
 */
abstract class UserAdapter(users:ArrayList<User> ): RecyclerView.Adapter<UserAdapter.ViewHolder>(){

    val mUsers = users

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.user_card, parent, false)
        return ViewHolder(itemView)
    }

    override fun getItemCount(): Int {
        return mUsers.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val user = mUsers[position]
        holder.userName.text = user.name
        holder.userAge.text = user.age.toString()

    }

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        var userName: TextView = view.findViewById<TextView>(R.id.user_name)
        var userAge: TextView = view.findViewById<TextView>(R.id.user_age)

    }

}